### PR TITLE
refactor! [BREAKING] Switching to Project paradigm - 2

### PR DESCRIPTION
**Pull request by JigarJoshi**

### CODE REVIEW

File: server/v1/api.proto [hunk 1]
```diff
@@ -816,22 +807,23 @@ service Tigris {
...
     };
     option(openapi.v3.operation) = {
       summary: "Create or update a collection"
       tags: "Collections"
     };
   }
-  // Drop the collection inside this project. This API deletes all of the documents inside this collection and any metadata associated with it.
-  rpc DropCollection(DropCollectionRequest) returns (DropCollectionResponse) {
+  // Delete the collection inside this project. This API deletes all of the documents inside this collection and any metadata associated with it.
+  rpc DeleteCollection(DeleteCollectionRequest) returns
```
Comment (himank) [inline server/v1/api.proto:819]: It is under database now, so should be fine to call it DropCollection

File: server/v1/api.proto [hunk 2]
```diff
@@ -445,8 +448,8 @@ message Page {
 message CreateProjectRequest {
   // Create project with this name.
   string project = 1;
-  ProjectOptions options = 2;
-  // Optionally specify a project branch name to perform operation on
+  DatabaseOptions options = 2;
```
Comment (himank) [inline server/v1/api.proto:451]: Why do we need it here?

File: server/v1/api.proto (hunk 1 above)
Comment (JigarJoshi) [reply server/v1/api.proto:819]: we can use single term for all delete operations so it remains consistent in CLI too.

wdyt?

Comment (himank) [reply server/v1/api.proto:819]: I think with scoping it doesn't matter. We can choose what makes more sense now from an individual operations perspective.

Comment (ovaistariq) [reply server/v1/api.proto:819]: With scoping, we don't need to use similar terms. Drop makes more sense.

File: server/v1/api.proto [hunk 3]
```diff
@@ -445,8 +448,8 @@ message Page {
...
+  // Optionally specify a database branch name to perform operation on
   string branch = 3;
```
Comment (ovaistariq) [inline server/v1/api.proto:453]: Branch should also not be needed when creating a project

File: server/v1/api.proto [hunk 4]
```diff
@@ -543,29 +544,27 @@ message CreateOrUpdateCollectionResponse {
...
   // A detailed response message.
   string message = 1;
   // An enum with value set as "dropped".
   string status = 2;
 }
 
 message ProjectInfo {
-  // Project name.
-  string project = 1;
+  // Database name.
```
Comment (ovaistariq) [inline server/v1/api.proto:563]: Why are you changing this here?

File: server/v1/api.proto [hunk 5]
```diff
@@ -543,29 +544,27 @@ message CreateOrUpdateCollectionResponse {
...
+  string database = 1;
 
-  // Metadata about the project.
-  ProjectMetadata metadata = 2;
+  // Metadata about the database.
+  DatabaseMetadata metadata = 2;
```
Comment (ovaistariq) [inline server/v1/api.proto:567]: Why the change? Project will have metadata as well.

File: server/v1/api.proto [hunk 6]
```diff
@@ -585,35 +584,35 @@ message ListProjectsResponse {
...
-  // Optionally specify a project branch name to perform operation on
+  // Optionally specify a database branch name to perform operation on
   string branch = 3;
 }
 
 message ListCollectionsResponse {
-  // Name of the project.
-  string project = 1;
+  // Name of the database.
+  string database = 1;
```
Comment (ovaistariq) [inline server/v1/api.proto:596]: Since Database is a meta concept, whats the purpose of returning database name?

File: server/v1/api.proto (hunk 1 above)
Comment (JigarJoshi) [reply server/v1/api.proto]: ok renamed it to DropCollection.

File: server/v1/api.proto (hunk 2 above)
Comment (JigarJoshi) [reply server/v1/api.proto:451]: So the thinking is - when you create project, we auto create DB for you and as part of that project creation operation, if you want to customize how your db gets created this is the option.

File: server/v1/api.proto (hunk 3 above)
Comment (JigarJoshi) [reply server/v1/api.proto]: makes sense.

File: server/v1/api.proto (hunk 5 above)
Comment (JigarJoshi) [reply server/v1/api.proto:567]: We can add other components's metadata as project grows. This represents for your project this is the metadata for your db. (which is part of project).

wdyt?

File: server/v1/api.proto (hunk 6 above)
Comment (JigarJoshi) [reply server/v1/api.proto:596]: Good point. Should we return project name instead?

File: server/v1/api.proto (hunk 2 above)
Comment (ovaistariq) [reply server/v1/api.proto:453]: What would be that customization? I think it is a little confusing here to add DatabaseOptions

File: server/v1/api.proto (hunk 4 above)
Comment (JigarJoshi) [reply server/v1/api.proto]: reverted.

File: server/v1/api.proto (hunk 5 above)
Comment (ovaistariq) [reply server/v1/api.proto:567]: OK. One thing I don't understand though is why we return databaseMetadata as part of dropCollectionResponse?

File: server/v1/api.proto (hunk 6 above)
Comment (ovaistariq) [reply server/v1/api.proto:596]: Why do we need to return the database / project name anyway as part of this response?

Comment (JigarJoshi) [reply server/v1/api.proto:596]: ok removed it from response.

File: server/v1/api.proto (hunk 5 above)
Comment (ovaistariq) [reply server/v1/api.proto:567]: Since we are returning project information, it would be better to return project metadata as there will be more things than database in a project.

File: scripts/fix_openapi.sh [hunk 7]
```diff
@@ -68,7 +68,7 @@ main() {
 	yq_del_service_tags
 
 	for i in InsertRequest ReplaceRequest UpdateRequest DeleteRequest ReadRequest \
-		CreateOrUpdateCollectionRequest DropCollectionRequest \
+		CreateOrUpdateCollectionRequest DropcCollectionRequest \
```
Comment (adilansari) [inline scripts/fix_openapi.sh:71]:
```suggestion
		CreateOrUpdateCollectionRequest DropCollectionRequest \
```

File: server/v1/api.proto [hunk 8]
```diff
@@ -544,13 +545,11 @@ message CreateOrUpdateCollectionResponse {
...
 message DropCollectionRequest {
-  // Project name of the collection.
+  // Project name whose db is under target to delete collection.
   string project = 1;
-  // Collection name to drop.
+  // Collection name to delete.
   string collection = 2;
   CollectionOptions options = 3;
-  // Optionally specify a project branch name to perform operation on
-  string branch = 4;
```
Comment (adilansari) [inline server/v1/api.proto:553]: Drop collection can be on a branch. Let's keep this in the meantime.

File: server/v1/api.proto [hunk 9]
```diff
@@ -626,26 +622,26 @@ message DescribeCollectionRequest {
...
 
-// A detailed description of the project and all the associated collections.
+// A detailed description of the database and all the associated collections.
 // Description of the collection includes schema details as well.
-message DescribeProjectResponse {
-  // Name of the project.
-  string project = 1;
+message DescribeDatabaseResponse {
+  // Name of the database.
+  string database = 1;
```
Comment (adilansari) [inline server/v1/api.proto:633]: I don't think this will be useful to the end user.

File: scripts/fix_openapi.sh (hunk 7 above)
Comment (JigarJoshi) [reply scripts/fix_openapi.sh]: Thanks.

File: server/v1/api.proto (hunk 5 above)
Comment (JigarJoshi) [reply server/v1/api.proto:567]: Ofcourse, we can extend it by adding project level thing at top level response once we have more to serve.

File: server/v1/api.proto (hunk 2 above)
Comment (JigarJoshi) [reply server/v1/api.proto:453]: So this is the only way for user to be able to create DB (by creating project). We always had this DatabaseOptions and so it got moved here.

Comment (JigarJoshi) [reply server/v1/api.proto:453]: We will also need project options here maybe wrap DatabaseOptions under ProjectOptions?

File: server/v1/api.proto (hunk 8 above)
Comment (JigarJoshi) [reply server/v1/api.proto]: 👍

File: server/v1/api.proto (hunk 9 above)
Comment (JigarJoshi) [reply server/v1/api.proto:635]: @efirs is it used in scaffolding?

File: server/v1/api.proto (hunk 2 above)
Comment (himank) [reply server/v1/api.proto:453]: Or remove it completely, if needed we can always add it later?

Comment (ovaistariq) [reply server/v1/api.proto:453]: Yes I vote for removing as well

File: server/v1/api.proto (hunk 5 above)
Comment (ovaistariq) [reply server/v1/api.proto:569]: Right but even if we have a single product right now `database`, I would assume that projectInfo has projectMetadata.

File: server/v1/api.proto (hunk 9 above)
Comment (ovaistariq) [reply server/v1/api.proto:635]: This is used by Webconsole for example to fetch all the schema for all the collections at once.

Comment (adilansari) [reply server/v1/api.proto:635]: I meant, the database name

Comment (ovaistariq) [reply server/v1/api.proto:635]: Ok, yeah database name isn't needed here

Comment (efirs) [reply server/v1/api.proto:635]: No. It's not used. We can remove.

File: server/v1/api.proto (hunk 2 above)
Comment (JigarJoshi) [reply server/v1/api.proto:453]: ok removed.

File: server/v1/api.proto (hunk 5 above)
Comment (JigarJoshi) [reply server/v1/api.proto:569]: renamed it to projectmetadata

File: server/v1/api.proto (hunk 9 above)
Comment (JigarJoshi) [reply server/v1/api.proto:635]: removed

**Review by himank — APPROVED** [review]
